python/lsst/summit/extras/logUtils.py

```diff
@@ -0,0 +1,265 @@
+                                                     collections=self.collection)
+        results = list(results)
+        self.log.info(f"Found {len(results)} datasets in collection for task {self.taskName}")
+        return sorted(results)
+
+    def _loadLogs(self, dataRefs):
+        """Load all the logs for the dataRefs.
+
+        Returns
+        -------
+        logs : `dict` {`lsst.daf.butler.core.datasets.ref.DatasetRef`:
+                       `lsst.daf.butler.core.logging.ButlerLogRecords`}
+            A dict of all the logs, keyed by their dataRef.
+        """
+        logs = {}
+        for i, dataRef in enumerate(dataRefs):
+            if (i+1) % 100 == 0:
+                self.log.info(f"Loaded {i+1} logs...")
+            log = self.butler.getDirect(dataRef)
+            logs[dataRef] = log
+        return logs
+
+    def getPassingDataIds(self):
+        """Get the dataIds for all passes within the collection for the task.
+
+        Returns
+        -------
+        dataIds : `list` [`lsst.daf.butler.dimensions.DataCoordinate`]
+        """
+        fails = self._getFailDataRefs()
+        passes = [r.dataId for r in self.dataRefs if r not in fails]
+        return passes
+
+    def getFailingDataIds(self):
+        """Get the dataIds for all fails within the collection for the task.
+
+        Returns
+        -------
+        dataIds : `list` [`lsst.daf.butler.dimensions.DataCoordinate`]
+        """
+        fails = self._getFailDataRefs()
+        return [r.dataId for r in fails]
+
+    def printPasses(self):
+        """Print out all the passing dataIds.
+        """
+        passes = self.getPassingDataIds()
+        for dataId in passes:
+            print(dataId)
+
+    def printFails(self):
+        """Print out all the failing dataIds.
+        """
+        passes = self.getFailingDataIds()
+        for dataId in passes:
+            print(dataId)
+
+    def countFails(self):
+        """Print a count of all the failing dataIds.
+        """
+        print(f"{len(self._getFailDataRefs())} failing cases found")
+
+    def countPasses(self):
+        """Print a count of all the passing dataIds.
+        """
+        print(f"{len(self.getPassingDataIds())} passing cases found")
+
+    def _getFailDataRefs(self):
+        """Get a list of all the failing dataRefs.
+
+        Note that these are dataset references to the logs, and as such are
+        not fails themselves, but logs containing the fail messages, and as
+        such the item of interest for the failures are their dataIds. This is
+        why ``_getFailDataRefs()`` is a private method, but getFailingDataIds
+        is the public API.
+
+        Returns
+        -------
+        logs : `list` [`lsst.daf.butler.core.datasets.ref.DatasetRef`]
+            A list of all the failing dataRefs.
+        """
+        fails = []
+        for dataRef, log in self.logs.items():
+            if log[-1].message.find('failed') != -1:
```

Comment:
Can you add a comment/additional docstring note about how this works?  `log[-1].message` looks a lot like magic to me, and looking up `DatasetRef` in `daf_butler` doesn't make it any clearer.

Reply:
Ah yes, very good point, that totally relies on understanding the logs. I will try to add a few comments around there then.